matchers/caseInsensitive/todomatcher.go

```diff
@@ -0,0 +1,39 @@
+package caseInsensitive
+
+import (
+	"regexp"
+
+	"github.com/preslavmihaylov/todocheck/matchers"
+)
+
+type TodoMatcherCaseInsensitive struct {
+	matcher matchers.TodoMatcher
+}
+
+func NewTodoMatcherCaseInsensitive(matcher matchers.TodoMatcher) *TodoMatcherCaseInsensitive {
+	return &TodoMatcherCaseInsensitive{
+		matcher: matcher,
+	}
+}
+
+func (m *TodoMatcherCaseInsensitive) IsMatch(expr string) bool {
+	return m.matcher.IsMatch(todoToUpper(expr))
+}
+
+func (m *TodoMatcherCaseInsensitive) IsValid(expr string) bool {
```

Comment:
The name of the struct should be `TodoMatcher` so that the usage is `case_insensitive.TodoMatcher` which looks more sleek